### PR TITLE
Remove MagicLeap support

### LINKED ISSUE #29773: Remove all magic leap embedding and support code
This includes:
* ports/libmlservo
* https://github.com/servo/servo/blob/4339b3bab494e477b502f82137bd360a963ec2be/python/servo/command_base.py#L813
* any python code using a `magicleap` argument
* https://github.com/servo/servo/tree/4339b3bab494e477b502f82137bd360a963ec2be/support/magicleap

### PR DESCRIPTION
**Pull request by mikayelgr**
Discussed in more detail in #29773.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes fix #29773 (GitHub issue number if applicable)

### CODE REVIEW

File: python/servo/command_base.py [hunk 1]
```diff
@@ -868,17 +859,13 @@ def build_like_command_arguments(decorated_function):
...
         if android is None:
             android = self.config["build"]["android"]
         if target and android:
             assert self.handle_android_target(target)
         if android and not target:
             target = self.config["android"]["target"]
-        if magicleap and not target:
-            target = "aarch64-linux-android"
-        if target and not android and not magicleap:
-            android = self.handle_android_target(target)
```
Comment (jdm) [inline python/servo/command_base.py:881]: These lines should stay; we should just remove the `and not magicleap` condition.

File: python/servo/command_base.py [hunk 2]
```diff
@@ -932,8 +919,6 @@ def run_cargo_build_like_command(
         if "-p" not in cargo_args:  # We're building specific package, that may not have features
             if self.config["build"]["debug-mozjs"] or debug_mozjs:
                 features.append("debugmozjs")
-            if not magicleap:
-                features.append("native-bluetooth")
```
Comment (jdm) [inline python/servo/command_base.py:936]: This should stay, but we should remove the `if not magicleap` condition.

**Comment by michaelgrigoryan25** [comment]
There are some references to MagicLeap in the Rust also:
- <https://github.com/servo/servo/blob/a30f9d6f17262bf1ef90481722624fd2b8d888a0/ports/libmlservo/src/lib.rs#L13>
- <https://github.com/servo/servo/blob/a30f9d6f17262bf1ef90481722624fd2b8d888a0/ports/libmlservo/src/lib.rs#L35>
- <https://github.com/servo/servo/blob/a30f9d6f17262bf1ef90481722624fd2b8d888a0/ports/libmlservo/src/lib.rs#L153-L165>

**Review by jdm — APPROVED** [review]
Looking good!

**Comment by jdm** [comment]
These tidy errors will need to be addressed, too:
```
./python/servo/package_commands.py:21: F401 'platform' imported but unused

./python/servo/package_commands.py:93: F811 redefinition of unused 'platform' from line 21

./python/servo/package_commands.py:476: F811 redefinition of unused 'platform' from line 21

./ports/libmlservo/Cargo.toml:28: no newline at EOF
```

File: ports/libmlservo/Cargo.toml [hunk 3]
```diff
@@ -25,4 +25,4 @@ servo-egl = "0.2"
 simpleservo = { path = "../libsimpleservo/api", features = ["no_static_freetype"] }
 smallvec = { workspace = true }
 webxr = { git = "https://github.com/servo/webxr", features = ["ipc", "magicleap"] }
-webxr-api = { git = "https://github.com/servo/webxr", features = ["ipc"] }
+webxr-api = { git = "https://github.com/servo/webxr", features = ["ipc"] }
```
Comment (jdm) [inline ports/libmlservo/Cargo.toml:28]: The libmlservo directory should be removed, as well as this line: https://github.com/servo/servo/blob/master/Cargo.toml#L8

**Comment by michaelgrigoryan25** [comment]
`test-tidy` doesn't report any issues from my side though 🤔

**Comment by michaelgrigoryan25** [comment]
I think we're done here.

**Comment by jdm** [comment]
@bors-servo r+
Thank you!

**Comment by michaelgrigoryan25** [comment]
Np!

**Pull request merged**